x/marker/types/msg_test.go

```diff
@@ -165,6 +165,105 @@ func TestMsgIbcTransferRequestValidateBasic(t *testing.T) {
 	}
 }
 
+// if len(msg.Manager) == 0 && msg.Status == StatusProposed {
+// 	return fmt.Errorf("marker manager cannot be empty when creating a proposed marker")
+// }
+// if msg.Status != StatusFinalized && msg.Status != StatusProposed {
+// 	return fmt.Errorf("marker can only be created with a Proposed or Finalized status")
+// }
+// testCoin := sdk.Coin{
+// 	Denom:  msg.Amount.Denom,
+// 	Amount: msg.Amount.Amount,
+// }
+// if !testCoin.IsValid() {
+// 	return fmt.Errorf("invalid marker denom/total supply: %w", sdkerrors.ErrInvalidCoins)
+// }
+
+func TestMsgAddMarkerRequestValidateBasic(t *testing.T) {
+	validAddress := sdk.MustAccAddressFromBech32("cosmos1sh49f6ze3vn7cdl2amh2gnc70z5mten3y08xck")
+
+	cases := []struct {
+		name     string
+		msg      MsgAddMarkerRequest
+		errorMsg string
+	}{
+		{
+			"should fail on attributes for non restricted coin",
+			*NewMsgAddMarkerRequest(
+				"hotdog",
+				sdk.NewInt(100),
+				validAddress,
+				validAddress,
+				MarkerType_Coin,
+				true,
+				true,
+				false,
+				[]string{"blah"},
+			),
+			"required attributes are reserved for restricted markers",
+		},
+		{
+			"should succeed on attributes for restricted coin",
+			*NewMsgAddMarkerRequest(
+				"hotdog",
+				sdk.NewInt(100),
+				validAddress,
+				validAddress,
+				MarkerType_RestrictedCoin,
+				true,
+				true,
+				false,
+				[]string{"blah"},
+			),
+			"",
+		},
+		{
+			"should succeed on for restricted coin",
+			*NewMsgAddMarkerRequest(
+				"hotdog",
+				sdk.NewInt(100),
+				validAddress,
+				validAddress,
+				MarkerType_RestrictedCoin,
+				true,
+				true,
+				false,
+				[]string{},
+			),
+			"",
+		},
+		{
+			"should succeed on for non-restricted coin",
+			*NewMsgAddMarkerRequest(
+				"hotdog",
+				sdk.NewInt(100),
+				validAddress,
+				validAddress,
+				MarkerType_Coin,
+				true,
+				true,
+				false,
+				[]string{},
+			),
+			"",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.msg.ValidateBasic()
+			if len(tc.errorMsg) > 0 {
+				assert.Error(t, err)
+				assert.Equal(t, tc.errorMsg, err.Error())
```

Comment:
These can be combined using `EqualError`:
```suggestion
				assert.EqualError(t, err, tc.errorMsg)
```